[refactor] Move patterns in QuantFusion to a separate file

jerryzh168 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #26939 [quant][graphmode] Integrate prepacked workaround in QuantFusion
* #26579 [quant][graphmode] Fold prepacked weight into module
* #26959 [quant][graphmode] Add InsertPackUnpack pass
* **#26848 [refactor] Move patterns in QuantFusion to a separate file**

Summary:
att

Test Plan:
ci

Reviewers:
pt1quant

Subscribers:

Tasks:

Tags:

Differential Revision: [D17636399](https://our.internmc.facebook.com/intern/diff/D17636399)

jerryzh168 committed:
[refactor] Move patterns in QuantFusion to a separate file

Summary:
att

Test Plan:
ci

Reviewers:
pt1quant

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

ZolotukhinM reviewed (changes requested):
With this change these strings will stay in memory forever even after fusion is finished, which is not great. Also, we are adding a lot of symbols with generic names like `addmm` to `torch::jit` namespace, I think that's also not good. How about adding a function that will return the patterns dictionary and moving this function to this new file? This way all strings will be stored in temporaries and we won't need any symbols in the namespace.

jerryzh168 committed:
Update on "[refactor] Move patterns in QuantFusion to a separate file"


Summary:
att

Test Plan:
ci

Reviewers:
pt1quant

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 commented:
do we want the strings to be static or just temporary?

ZolotukhinM commented:
> do we want the strings to be static or just temporary?

I would prefer temporary, what's the point of keeping them around?

jerryzh168 commented:
> > do we want the strings to be static or just temporary?
> 
> I would prefer temporary, what's the point of keeping them around?

so that no need to reconstruct them every time people call QuantFusion?

jerryzh168 committed:
Update on "[refactor] Move patterns in QuantFusion to a separate file"


Summary:
att

Test Plan:
ci

Reviewers:
pt1quant

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 commented:
updated with temporary

jerryzh168 committed:
Update on "[refactor] Move patterns in QuantFusion to a separate file"


Summary:
att

Test Plan:
ci

Reviewers:
pt1quant

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

ZolotukhinM commented on torch/csrc/jit/passes/quantization_patterns.h:
```diff
@@ -0,0 +1,77 @@
+#pragma once
+
```
Nit: please add
```
#include <string>
#include <unordered_map>
```
Otherwise if the include order in `quantization.cpp` changes for some reason it might fail to build. Also, might be worth considering making it in a cpp file (vs in a .h).

jerryzh168 replied:
I'll move to cpp in a separate PR since this one is in the bottom of a stack

ZolotukhinM commented on torch/csrc/jit/passes/quantization.cpp:
```diff
@@ -1,4 +1,5 @@
 #include <torch/csrc/jit/passes/quantization.h>
+#include <torch/csrc/jit/passes/quantization_patterns.h>
```
Nit: I suspect clang-format would put this include lower.

ZolotukhinM reviewed (approved):
👍

jerryzh168 committed:
Update on "[refactor] Move patterns in QuantFusion to a separate file"


Summary:
att

Test Plan:
ci

Reviewers:
pt1quant

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[refactor] Move patterns in QuantFusion to a separate file"


Summary:
att

Test Plan:
ci

Reviewers:
pt1quant

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[refactor] Move patterns in QuantFusion to a separate file"


Summary:
att

Test Plan:
ci

Reviewers:
pt1quant

Subscribers:

Tasks:

Tags:

Differential Revision: [D17636399](https://our.internmc.facebook.com/intern/diff/D17636399)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[refactor] Move patterns in QuantFusion to a separate file"


Summary:
att

Test Plan:
ci

Reviewers:
pt1quant

Subscribers:

Tasks:

Tags:

Differential Revision: [D17636399](https://our.internmc.facebook.com/intern/diff/D17636399)

[ghstack-poisoned]
